setup.py

```diff
@@ -2,17 +2,20 @@
 from setuptools import setup
 import sys
 import subprocess as subp
+import platform
+import os
 
 cwd = Path(__file__).parent
 
 sys.path.append(str(cwd))
 from cmake_ext import CMakeExtension, CMakeBuild  # noqa: E402
 
-if (cwd / ".git").exists():
-    # make sure that submodules are up-to-date,
-    # it is a common error to forget this when
-    # switching between development branches
-    subp.check_call(["git", "submodule", "update"])
+if not os.environ.get("CI", False):
+    if (cwd / ".git").exists():
+        # make sure that submodules are up-to-date,
+        # it is a common error to forget this when
+        # switching between development branches
+        subp.check_call(["git", "submodule", "update"])
```

Comment:
This is ok, but it is less error prone to just replace `check_call` with `call` and remove the environ check.

Reply:
Ok. `check_call` is replaced with `call`.

Reply:
Peanut gallery comment: `run` is the modern (Python 3.5+) version of `call`/`check_call`/`check_output`. I'd (and the Python docs) recommend just always using `run` and ignoring the old API.